zhaquirks/tuya/ts0601_valve.py

```diff
@@ -471,6 +502,155 @@ class GiexIrrigationStatus(t.enum8):
 )
 
 
+# NEO NAS-WV03B
+(
+    TuyaQuirkBuilder("_TZE204_rzrrjkz2", "TS0601")
+    .applies_to("_TZE204_uab532m0", "TS0601")
+    .applies_to("_TZE204_z7a2jmyy", "TS0601")
+    .tuya_onoff(dp_id=1)
+    .tuya_enum(
+        dp_id=3,
+        attribute_name="valve_status",
+        enum_class=TuyaValveStatus,
+        translation_key="valve_status",
+        fallback_name="Valve status",
+    )
+    .tuya_number(
+        dp_id=5,
+        attribute_name="valve_countdown",
+        type=t.uint16_t,
+        device_class=SensorDeviceClass.DURATION,
+        unit=UnitOfTime.MINUTES,
+        min_value=1,
+        max_value=240,
+        step=1,
+        translation_key="valve_countdown",
+        fallback_name="Irrigation time",
+    )
+    .tuya_sensor(
+        dp_id=6,
+        attribute_name="valve_duration",
+        type=t.uint32_t,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DURATION,
+        unit=UnitOfTime.MINUTES,
+        entity_type=EntityType.STANDARD,
+        translation_key="valve_duration",
+        fallback_name="Irrigation duration",
+    )
+    .tuya_dp(
+        dp_id=9,
+        ep_attribute=TuyaValveWaterConsumedInstDemand.ep_attribute,
+        attribute_name=Metering.AttributeDefs.instantaneous_demand.name,
+    )
+    .tuya_metering(dp_id=15, metering_cfg=TuyaValveWaterConsumedInstDemand)
+    .tuya_battery(dp_id=11, battery_type=BatterySize.AA, battery_qty=2)
+    .tuya_binary_sensor(
+        dp_id=19,
+        attribute_name="valve_fault",
+        translation_key="valve_fault",
+        fallback_name="Valve fault",
+    )
+    .tuya_enum(
+        dp_id=37,
+        attribute_name="weather_delay",
+        enum_class=TuyaValveWeatherDelayVar02,
+        translation_key="weather_delay",
+        fallback_name="Weather delay",
+        initially_disabled=True,
+    )
+    # Charstring normal timed, dp 38 omitted
+    .tuya_switch(
+        dp_id=42,
+        attribute_name="switch_enabled",
+        entity_type=EntityType.STANDARD,
+        translation_key="switch_enabled",
+        fallback_name="Switch enabled",
+    )
+    .tuya_sensor(
+        dp_id=47,
+        attribute_name="smart_irrigation",
+        type=t.uint16_t,
+        entity_type=EntityType.DIAGNOSTIC,
+        translation_key="smart_irrigation",
+        fallback_name="Smart irrigation",
+    )
+    .tuya_switch(
+        dp_id=101,
+        attribute_name="total_flow_reset_switch",
+        entity_type=EntityType.STANDARD,
+        translation_key="total_flow_reset_switch",
+        fallback_name="Total flow reset switch",
+    )
```

Comment:
Is it clear if this resets automatically? If so, it could also be a button I guess.
Although we might need to add support for `tuya_button` first..

Reply:
It's not clear, z2m exposes this as a binary input so I matched that.